datalad/support/gitrepo.py

```diff
@@ -2117,6 +2117,8 @@ def checkout(self, name, options=None):
         cmd += [str(name)]
 
         self._git_custom_command('', cmd, expect_stderr=True)
+        # Note: force=True shouldn't be needed, since mtime changes on checkout:
+        self.config.reload()
```

Comment:
If `.config` is not detecting the change (I thought it does!?) I wonder if to be safe we should just do it for every `_git_custom_command`?  or at least also in those other which adjust current worktree: `merge`, `cherry_pick`, `pull`, etc?

Reply:
It does detect based on mtime. But the detection happens within `reload` (if not called with `force=True`). Where else?

Yes, other methods, that might change `.datalad/config` should also call `reload`.

Reply:
So, shouldn't this be done here?  and if based on mtime I would just call in each `_git_custom_command` to avoid duplication etc

Reply:
Well, it could be done within `_git_custom_command` but that would be way more expensive, than figuring out which methods really need it. What's the point of checking mtimes of config files after calling `git ls-files`?

Reply:
I think we should make it more declarative/centralized because may be other things would need to be done in such cases besides reloading config.  I would have added `updates_tree=False` kwarg to the `_git_custom_command`, and then added the logic to call this reload in there conditioned on that argument.  Necessary methods would then specify `updates_tree=True` where applicable.

Reply:
Ok. Will do.